From 186f82627c40d0c3a56a6a94ce55c055ad1f7620 Mon Sep 17 00:00:00 2001
From: George Joseph <gjoseph@digium.com>
Date: Fri, 10 Nov 2017 09:26:29 -0700
Subject: [PATCH] sip_parser: Fix return code in pjsip_find_msg and add
 "volatile"

The default return code for pjsip_find_msg was PJ_SUCCESS so if
a Content-Length header wasn't found at all, pjsip_find_msg was
returning PJ_SUCCESS instead of PJSIP_EMISSINGHDR.

Also added the volatile keyword to a few variables what are used
both insude and outsude the PJ_TRY/PJ_CATCH block.
---
 pjsip/src/pjsip/sip_parser.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/pjsip/src/pjsip/sip_parser.c b/pjsip/src/pjsip/sip_parser.c
index f9a0e65b5..f65ffca5a 100644
--- a/pjsip/src/pjsip/sip_parser.c
+++ b/pjsip/src/pjsip/sip_parser.c
@@ -834,13 +834,13 @@ PJ_DEF(pj_status_t) pjsip_find_msg( const char *buf, pj_size_t size,
 				  pj_bool_t is_datagram, pj_size_t *msg_size)
 {
 #if PJ_HAS_TCP
-    const char *hdr_end;
-    const char *body_start;
+    const char *volatile hdr_end;
+    const char *volatile body_start;
     const char *pos;
-    const char *line;
+    const char *volatile line;
     int content_length = -1;
     pj_str_t cur_msg;
-    pj_status_t status = PJ_SUCCESS;
+    volatile pj_status_t status = PJSIP_EMISSINGHDR;
     const pj_str_t end_hdr = { "\n\r\n", 3};
 
     *msg_size = size;
-- 
2.13.6

